client/CPlayerInterface.cpp

```diff
@@ -192,24 +193,39 @@ void CPlayerInterface::playerStartsTurn(PlayerColor player)
 
 void CPlayerInterface::performAutosave()
 {
-	std::string prefix = settings["session"]["saveprefix"].String();
 	int frequency = static_cast<int>(settings["general"]["saveFrequency"].Integer());
-	if(firstCall)
+	if(frequency > 0 && cb->getDate() % frequency == 0)
 	{
-		autosaveCount = getLastIndex(prefix + "Autosave_");
+		bool usePrefix = settings["general"]["useSavePrefix"].Bool();
+		std::string prefix = std::string();
 
-		if(firstCall > 0) //new game, not loaded
+		if(usePrefix)
 		{
-			int index = getLastIndex(prefix + "Newgame_");
-			index %= SAVES_COUNT;
-			cb->save("Saves/" + prefix + "Newgame_Autosave_" + std::to_string(index + 1));
+			prefix = settings["general"]["savePrefix"].String();
+			if(prefix.empty())
+			{
+				prefix = cb->getMapHeader()->name.substr(0, 5) + "_";
+			}
+		}
+
+		autosaveCount++;
+
+		int autosaveCountLimit = settings["general"]["autosaveCountLimit"].Integer();
+		if(autosaveCountLimit > 0)
+		{
+			cb->save("Saves/" + prefix + "Autosave_" + std::to_string(autosaveCount));
+			autosaveCount %= autosaveCountLimit;
+		}
+		else
+		{
+			std::string stringifiedDate = (cb->getDate(Date::MONTH) < 10
+				? std::string("0") + std::to_string(cb->getDate(Date::MONTH))
+				: std::to_string(cb->getDate(Date::MONTH)))

```

Comment:
Not sure if such conditional is needed. "123" form is easily recognizeable, but "0123" is not.

I mean, I get that this is to make sorting look good once you reach month 10, but same can be said about reaching month 100.

I'd rather consider altering sorting to account for integer numbering